File: PlayersApp/src/api/apiClient.ts
import axios from 'axios';
import { apiConfig } from '../config/config';
import { PaginatedItems } from '../types/Player';

const apiClient = axios.create({
  baseURL: apiConfig.baseURL,
  timeout: apiConfig.timeout,
});

export const getPlayers = async (
  page: number,
  limit: number,
  searchQuery: string,
  selectedPosition: string
): Promise<PaginatedItems<Player>> => {
  const { data } = await apiClient.get(`/players`, {
    params: {
      page,
      limit,
      search: searchQuery,
      position: selectedPosition,
    },
  });
  return data;
};

export default apiClient;

File: PlayersNativeiOSApp/PlayersDemo/Api/ApiClient.swift
import Foundation

struct APIClient {
    static let shared = APIClient()
    
    func getPlayers(page: Int, limit: Int, searchQuery: String, selectedPosition: String) async throws -> PaginatedItems<Player> {
        let queryItems = [
            URLQueryItem(name: "page", value: "\(page)"),
            URLQueryItem(name: "limit", value: "\(limit)"),
            URLQueryItem(name: "search", value: searchQuery),
            URLQueryItem(name: "position", value: selectedPosition)
        ]
        
        guard let url = APIUtils.makeURL(baseURL: Config.apiBaseURL, path: "/players", queryItems: queryItems) else {
            throw APIClientError.invalidURL
        }
        
        var request = URLRequest(url: url)
        request.timeoutInterval = Config.requestTimeout
        
        do {
            let (data, response) = try await URLSession.shared.data(for: request)
            try APIUtils.handleResponse(response)
            return try APIUtils.decode(data, to: PaginatedItems<Player>.self)
        } catch let error as URLError where error.code == .timedOut {
            throw APIClientError.timeout
        } catch {
            throw error
        }
    }
}
